### PR TITLE
Jaeger Version++ && Version documentation

### PR DESCRIPTION
**Pull request by Aergonus**
Similar to #70, addresses #88

Signed-off-by: Victor Lei <KLEI22@bloomberg.net>

### CODE REVIEW

**Comment by yurishkuro** [comment]
> using your real name (sorry, no pseudonyms or anonymous contributions.)

**Comment by Spellchaser** [comment]
Heh I see DCO pulls from the git config.

**Comment by Spellchaser** [comment]
Heh... I see. DCO pulls from git config. 
https://github.com/probot/dco/issues/43

**Comment by Spellchaser** [comment]
It seems like your jenkin jobs are a bit broken.
The only change in the commit from before aka [#254](https://travis-ci.org/jaegertracing/jaeger-kubernetes/builds/382302837) is the commit sign-off, yet it failed CI on the next run [#255](https://travis-ci.org/jaegertracing/jaeger-kubernetes/builds/382363652).

**Comment by yurishkuro** [comment]
>  I see. DCO pulls from git config. probot/dco#43

I don't think DCO has access to your git config. I am not sure how in that ticket the Author is different from the sign-off, wouldn't they both come from git config?

Personally I recommend using your full name in the GitHub profile, I think it takes care of the issue. It's not like your full name is a secret if you're putting it in the commit signature anyway.

File: README.md [hunk 1]
```diff
@@ -147,6 +147,10 @@ kubectl run jaeger-spark-dependencies --schedule="55 23 * * *" --env="STORAGE=ca
 
 If you want to run the job only once and immediately then remove scheduled flag.
 
+## Docker Versions
+The jaeger project automatically creates new docker images with versions that mirror the release number. Feel free to pick the latest release.
```
Comment (jpkrohling) [inline README.md]: s/jaeger/Jaeger

Comment (jpkrohling) [reply README.md]: s/docker/Docker

File: README.md [hunk 2]
```diff
@@ -147,6 +147,10 @@ kubectl run jaeger-spark-dependencies --schedule="55 23 * * *" --env="STORAGE=ca
 
 If you want to run the job only once and immediately then remove scheduled flag.
 
+## Docker Versions
+The jaeger project automatically creates new docker images with versions that mirror the release number. Feel free to pick the latest release. 
+> A general k8 tip; it's recommended that you do not use `:latest` in production but rather pin the latest version.
```
Comment (jpkrohling) [inline README.md]: k8? Did you mean Kubernetes (k8s)? I think this tip is more general than that, so: 

> Docker pro-tip: do not use `:latest` in production, but rather, pin it to the latest release

Comment (Spellchaser) [reply README.md]: Correct the tags are parsed by docker, but seeing as this is a kubernetes repo and most people are deploying jaeger with kubernetes,  tagging this as a kubernetes/deployment related tip would be less confusing.

**Comment by Spellchaser** [comment]
ping!

File: jaeger-production-template.yml [hunk 3]
```diff
@@ -12,6 +12,10 @@
 # the License.
 #
 
+#
+# Check https://github.com/jaegertracing/jaeger/releases for the latest version to use
```
Comment (pavolloffay) [inline jaeger-production-template.yml]: can you please remove this?

File: jaeger-production-template.yml [hunk 4]
```diff
@@ -33,7 +37,7 @@ items:
           jaeger-infra: collector-pod
       spec:
         containers:
-        - image: jaegertracing/jaeger-collector:1.2
+        - image: jaegertracing/jaeger-collector:1.4.1
```
Comment (pavolloffay) [inline jaeger-production-template.yml]: The newest is 1.5.0

File: production/cassandra.yml [hunk 5]
```diff
@@ -120,7 +120,8 @@ items:
       spec:
         containers:
         - name: jaeger-cassandra-schema
-          image: jaegertracing/jaeger-cassandra-schema:1.2
+          image: jaegertracing/jaeger-cassandra-schema:1.4.1
+          # Check https://github.com/jaegertracing/jaeger/releases for the latest version
```
Comment (pavolloffay) [inline production/cassandra.yml]: The comment is not needed

**Comment by Spellchaser** [comment]
Right so the pinned versions here are seldom updated. For any new user coming to the project, it's better to at least have a comment warning them that this may not be the latest version. 

I spent more time than I'd like to admit on trying to get `base-path` to work when it was changed to `prefix`.

**Comment by pavolloffay** [comment]
> Right so the pinned versions here are seldom updated. For any new user coming to the project, it's better to at least have a comment warning them that this may not be the latest version.

I don't see such a comment in the readme. It would be better to add a sentence to Development setup about using latest and somewhere to production setup about using pinned version.

File: jaeger-production-template.yml (hunk 3 above)
Comment (pavolloffay) [reply jaeger-production-template.yml]: This should rather link to dockerhub but I think everybody knows what registry it is using and where to find more info

**Comment by Spellchaser** [comment]
The process for determining what version is useable for production seems to be the latest release for jaeger. Are you relying on the CI jobs to test the new versions for production?

**Comment by pavolloffay** [comment]
The pinned version is used to not accidentally break people on new releases. I wanted also tag this repo to version templates too as there can be changes too.

**Comment by Spellchaser** [comment]
Uhhh lemme change the README

**Comment by pavolloffay** [comment]
np I will wait

**Comment by Spellchaser** [comment]
I think that's valid english :) 
I figured that you didn't want that line in the docker images part

**Pull request merged**